lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOTeamDAOImpl.java

```diff
@@ -652,4 +656,11 @@ public List<String> getAllTeamsUserIsAdmin(String userId) {
 	public void truncateAll() {
 		jdbcTemplate.update("DELETE FROM " + TABLE_TEAM);
 	}
+
+	@Override
+	public TeamState getState(String teamId) throws DatastoreException, NotFoundException {
+		MapSqlParameterSource param = new MapSqlParameterSource();
+		param.addValue(COL_TEAM_ID, teamId);
+		return namedJdbcTemplate.queryForObject(SELECT_TEAM_STATE, param, TeamState.class);
```

Comment:
This would throw a EmptyResultDataAccessException if the query returned no rows, we can catch that and throw a NotFoundException